### DISCUSSION TITLE
Why vulnerabilityAlerts does not work in this case?

### DISCUSSION THREAD

**Discussion by unarist** [renovatebot/renovate#13588] [Request Help] [answered]
I found the `vulnerabilityAlerts` feature does not create a PR when...

* the `package.json` file is disabled by `packageRules` and
* `^` mark is used in the version specifier

The question is, why presence of `^` mark effects the behavior?
(I have expected `vulnerabilityAlerts` creates a PR even if the file is disabled)

## Base settings

renovate.json:
```
{
  "timezone": "Asia/Tokyo",
  "schedule": ["every 3 months on the first day of the month"],
  "dependencyDashboard": true,
  "vulnerabilityAlerts": {
    "enabled": true
  }
}
```

package.json:
```
{
  "dependencies": {
    "marked": "2.1.3"
  }
}
```

## Results

| package.json | file is disabled? [1] | PR created? |
|---|---|---|
| marked: 2.1.3 | Yes | Yes |
| marked: 2.1.3 | No | Yes |
| marked: ^2.1.3 | Yes | **No** |
| marked: ^2.1.3 | No | Yes |

[1] "file is disabled" means below settings in renovate.json:

```
  "packageRules": [
    {
      "matchFiles": ["package.json"],
      "enabled": false
    }
  ],
```

I have tested on https://github.com/unarist/renovate_secup_test.

**Comment by viceice** [comment, marked as answer]
Disabled is disabled. So all update types are disabled. You need another package rule to enable vulnerability alert update PR

**Comment by rarkins** [comment]
The short answer is: don't disable your package files if you want them remediated, or don't use ranges.

**Reply by unarist** [reply]
Okay, I understand disabling package files also disables vulnerabilityAlerts.

However, why using ranges change the behavior? It still confuses me.

**Reply by rarkins** [reply]
It's simply luck that it works at all in the other case. If you can't sleep at night without knowing exactly why, you're free to read the source code in this repo
